be/src/olap/storage_engine.cpp

```diff
@@ -1026,38 +1026,8 @@ void StorageEngine::notify_listeners() {
 }
 
 Status StorageEngine::execute_task(EngineTask* task) {
-    auto lock_related_tablets = [&]() -> std::vector<std::unique_lock<std::shared_mutex>> {
-        // add write lock to all related tablets
```

Comment:
task execute will modify tablet's meta, so that I think we should acquire lock here.

Reply:
No engine task has implement get_related_tablets() so it will always return empty, so no locks will acquire here